src/server/game/Handlers/GroupHandler.cpp

```diff
@@ -1134,3 +1134,69 @@ void WorldSession::HandleOptOutOfLootOpcode(WorldPacket& recvData)
 
     GetPlayer()->SetPassOnGroupLoot(passOnLoot);
 }
+
+void WorldSession::HandleGroupSwapSubGroupOpcode(WorldPacket& recv_data)
+{
+    std::string playerName1, playerName2;
+
+    recv_data >> playerName1;
+    recv_data >> playerName2;
+
+    if (!normalizePlayerName(playerName1))
+    {
+        SendPartyResult(PARTY_OP_SWAP, playerName1, ERR_GROUP_SWAP_FAILED);
+        return;
+    }
+    if (!normalizePlayerName(playerName2))
+    {
+        SendPartyResult(PARTY_OP_SWAP, playerName1, ERR_GROUP_SWAP_FAILED);
+        return;
+    }
+
+    Group* group = GetPlayer()->GetGroup();
+    if (!group || !group->isRaidGroup())
+        return;
+
+    if (!group->IsLeader(GetPlayer()->GetGUID()) &&
+        !group->IsAssistant(GetPlayer()->GetGUID()))
+        return;
+
+    //get guid, member may be offline
+    auto getGuid = [&group](std::string const& playerName)
```

Comment:
nice that you use `auto`, consider using it also in other places ;)

Reply:
I just back port the code as-is (as long as it works and it is an improvement), if you find something that can be optimized just leave a comment just like now and I will rewrite it.